[radio-group] Disabled state for individual buttons gets overridden in Lit version

### Description

The following ITs are failing when switching the Flow counterpart to the Lit based version:

```
[ERROR]   DisabledItemsPageIT.disabled_items_with_item_enabled_provider_should_stay_disabled_after_adding_renderer_and_enabling_group:95 Second button should be disabled expected:<true> but was:<null>
[ERROR]   DisabledItemsPageIT.disabled_items_with_item_enabled_provider_should_stay_disabled_after_enabling_group:76 Second button should be disabled expected:<true> but was:<null>
```

The error can be reproduced in the web component by changing `disabled` on the  `vaadin-radio-group` from `true` to `false` and immediately adding a `vaadin-radio-button` with `disabled` property set afterwards.

This only happens in Lit version since the `disabled` property does not have `sync: true` and causes async update.

### Expected outcome

Expected the following test to pass:

```js

it('should not override disabled property on dynamically added radio buttons', async () => {
  group.disabled = false;
  const radio = document.createElement('vaadin-radio-button');
  radio.disabled = true;
  group.appendChild(radio);
  await nextFrame();
  expect(radio.disabled).to.be.true;
});
```

### Minimal reproducible example

```html
<vaadin-radio-group label="Travel class" disabled>
  <vaadin-radio-button value="economy" label="Economy"></vaadin-radio-button>
  <vaadin-radio-button value="business" label="Business"></vaadin-radio-button>
  <vaadin-radio-button value="firstClass" label="First Class"></vaadin-radio-button>
</vaadin-radio-group>

<script type="module">
  import '@vaadin/radio-group/vaadin-lit-radio-group.js';

  const group = document.querySelector('vaadin-radio-group');
  await group.updateComplete;

  const radio = document.createElement('vaadin-radio-button');
  radio.value = 'plus';
  radio.label = 'Economy Plus';
  radio.disabled = true;


  group.disabled = false;
  group.appendChild(radio);
</script>
```

### Steps to reproduce

Run the test above, or check the HTML snippet in the dev page.

### Environment

Vaadin version(s): 24.6 alpha

### Browsers

Issue is not browser related

--- a/packages/a11y-base/src/disabled-mixin.js
+++ b/packages/a11y-base/src/disabled-mixin.js
@@ -23,6 +23,7 @@ export const DisabledMixin = dedupingMixin(
             value: false,
             observer: '_disabledChanged',
             reflectToAttribute: true,
+            sync: true,
           },
         };
       }

--- a/packages/a11y-base/test/disabled-mixin.test.js
+++ b/packages/a11y-base/test/disabled-mixin.test.js
@@ -1,23 +1,22 @@
 import { expect } from '@vaadin/chai-plugins';
-import { fixtureSync } from '@vaadin/testing-helpers';
+import { defineLit, definePolymer, fixtureSync, nextRender } from '@vaadin/testing-helpers';
 import sinon from 'sinon';
-import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';
+import { ControllerMixin } from '@vaadin/component-base/src/controller-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
 import { DisabledMixin } from '../src/disabled-mixin.js';
 
-customElements.define(
-  'disabled-mixin-element',
-  class extends DisabledMixin(PolymerElement) {
-    static get template() {
-      return html`<slot></slot>`;
-    }
-  },
-);
+const runTests = (defineHelper, baseMixin) => {
+  const tag = defineHelper(
+    'disabled-mixin',
+    '<slot></slot>',
+    (Base) => class extends DisabledMixin(baseMixin(Base)) {},
+  );
 
-describe('disabled-mixin', () => {
   let element;
 
-  beforeEach(() => {
-    element = fixtureSync(`<disabled-mixin-element></disabled-mixin-element>`);
+  beforeEach(async () => {
+    element = fixtureSync(`<${tag}></${tag}>`);
+    await nextRender();
   });
 
   it('should set disabled property to false by default', () => {
@@ -47,4 +46,12 @@ describe('disabled-mixin', () => {
     element.click();
     expect(spy.called).to.be.false;
   });
+};
+
+describe('DisabledMixin + Polymer', () => {
+  runTests(definePolymer, ControllerMixin);
+});
+
+describe('DisabledMixin + Lit', () => {
+  runTests(defineLit, PolylitMixin);
 });

--- a/packages/radio-group/test/radio-group.common.js
+++ b/packages/radio-group/test/radio-group.common.js
@@ -61,13 +61,12 @@ describe('radio-group', () => {
       buttons = [...group.querySelectorAll('vaadin-radio-button')];
     });
 
-    it('should propagate disabled property to radio buttons', async () => {
+    it('should propagate disabled property to radio buttons', () => {
       buttons.forEach((button) => {
         expect(button.disabled).to.be.true;
       });
 
       group.disabled = false;
-      await nextUpdate(group);
       buttons.forEach((button) => {
         expect(button.disabled).to.be.false;
       });
@@ -75,6 +74,15 @@ describe('radio-group', () => {
 
     it('should set disabled property to dynamically added radio buttons', async () => {
       const radio = document.createElement('vaadin-radio-button');
+      group.appendChild(radio);
+      await nextFrame();
+      expect(radio.disabled).to.be.true;
+    });
+
+    it('should not override disabled property on dynamically added radio buttons', async () => {
+      group.disabled = false;
+      const radio = document.createElement('vaadin-radio-button');
+      radio.disabled = true;
       group.appendChild(radio);
       await nextFrame();
       expect(radio.disabled).to.be.true;

